### PR TITLE
Remove file descriptor overload. 

### PR DESCRIPTION
**Pull request by FeepingCreature**
Error handling will be solved by manually iterating over loggers.
Changed because downstream loggers may have different layout requirements, which makes a single `fileDescriptor` approach unviable anyways.
Instead, expose the configured loggers.

### CODE REVIEW

**Comment by FeepingCreature** [comment]
Methods are prefixed with `__` to indicate that you are doing a very weird and dangerous thing when you call them.

File: src/util/log.d [hunk 1]
```diff
@@ -92,32 +92,20 @@ struct Log
...
         this.loggers = loggers.dup;
-        buffer = new int[this.loggers.length];
         levels = reduce!((a, b) => a | b.levels)(0, this.loggers);
     }
 
-    public int[] fileDescriptors() @nogc nothrow @safe
-    in (buffer.length >= loggers.length)
+    // Backdoor access for maintenance operations such as segfault handling.
+    // See `FileLogger.__fileDescriptor`.
+    public Logger[] __loggers() @nogc nothrow pure @safe
```
Comment (linkrope) [inline src/util/log.d:106]: This function just returns the loggers provided to the constructor:
how about saving them at the caller?

File: src/util/log.d [hunk 2]
```diff
@@ -345,9 +323,12 @@ class FileLogger(Layout) : Logger
         file.flush;
     }
 
-    override Nullable!int fileDescriptor() const @nogc nothrow @safe
+    // Segfault handling requires limiting ourselves to signal-safe functions,
+    // which is a very limited subset - Phobos/toString/anything GC is right out.
+    // For such cases, offer backdoor access to the file handle.
+    public int __fileDescriptor() const @nogc nothrow @safe
```
Comment (linkrope) [inline src/util/log.d:329]: I would just omit the leading underscores.
Why should a file logger not make its file descriptor readable?

**Review by linkrope — CHANGES_REQUESTED** [review]
At least, remove the underscores!

File: src/util/log.d (hunk 1 above)
Comment (FeepingCreature) [reply src/util/log.d:106]: Also works, but runs the risk of desyncing if for some reason `log` is changed again.

File: src/util/log.d (hunk 2 above)
Comment (FeepingCreature) [reply src/util/log.d:329]: No prob. I wanted to avoid these APIs being casually used. But I guess it's not an issue.

**Comment by FeepingCreature** [comment]
removed underscores

**Review by linkrope — CHANGES_REQUESTED** [review]

File: src/util/log.d [hunk 3]
```diff
@@ -189,7 +177,7 @@ struct Log
         eventInfo.file = file;
         eventInfo.line = line;
 
-        foreach (logger; loggers)
+        foreach (logger; loggers_)
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d:180]: done

**Review by linkrope — APPROVED** [review]

**Pull request merged**